File: src/main/java/ar/edu/utn/frba/tacs/tp/api/herocardsgame/models/game/Deck.java
package ar.edu.utn.frba.tacs.tp.api.herocardsgame.models.game;

import java.util.List;

public class Deck {

    private Long id;
    private String name;
    private List<Card> cards;

    public Deck(Long id, String name, List<Card> cards) {
        this.id = id;
        this.name = name;
        this.cards = cards;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public List<Card> getCards() {
        return cards;
    }
}

File: src/main/kotlin/ar/edu/utn/frba/tacs/tp/api/herocardsgame/models/game/Deck.kt
package ar.edu.utn.frba.tacs.tp.api.herocardsgame.models.game

data class Deck(
    val id: Long? = null,
    val name: String,
    val cards: List<Card> = emptyList(),
    val usable: Boolean = true
) {

    fun mixCards(): Deck = copy(cards = cards.shuffled())

    fun updateDeck(name: String?, cards: List<Card>): Deck =
        Deck(name = name ?: this.name, cards = cards.ifEmpty { this.cards })

}